lib/puppet/provider/firewall/iptables.rb

```diff
@@ -472,7 +474,7 @@ def self.rule_to_hash(line, table, counter)
     # Convert booleans removing the previous cludge we did
     @known_booleans.each do |bool|
       if hash[bool] != nil then
-        if hash[bool] != "true" then
+        if hash[bool] != "true" && hash[bool] != "!" then
```

Comment:
Yeah I did wonder if this is worth refactoring more of the code now.  From what I can make out the pre-cludging changes the args e.g.

```! --physdev-is-bridged``` to ```--physdev-is-bridged "!"```  or ```! --state INVALID``` to ```--state "! INVALID"```

Which works for state because it's just basic property however physdev-is-bridged is marked as a boolean so before it would've ended up been hard-coded to a string "true" it now picks up the quoted value in this case "!".

I'll see if I can refactor this a bit more or just add a test - if you have any particular way you'd like this handled let me know.